main_pullrequest.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/go-github/v28/github"
+
+	clientgithub "github.com/mendersoftware/integration-test-runner/client/github"
+)
+
+func processGitHubPullRequest(ctx *gin.Context, pr *github.PullRequestEvent, githubClient clientgithub.Client, conf *config) error {
+	log := getCustomLoggerFromContext(ctx)
+
+	// Do not run if the PR is a draft
+	if pr.GetPullRequest().GetDraft() {
+		log.Infof("The PR: %s/%d is a draft. Do not run tests", pr.GetRepo().GetName(), pr.GetNumber())
+		return nil
+	}
+
+	if isDependabotPR, err := maybeVendorDependabotPR(log, pr, conf); isDependabotPR {
```

Comment:
Question:
Is the `isDependabotPR` check necessary. Could it not be more like a:
```
if _, err := maybeVendorDependabotPR(log, pr, conf); err != nil {
...
```
it seems more like no-op in the no-err case.

Reply:
I don't think so; if isDependabotPR is true, we need to stop there; if we get an error, we log it and stop here
I don't think isDependabotPR can be true in case of error, so I adjusted a bit the if condition.
Does it make sense?